docs/blog/_posts/2017-10-02-fourth-dotty-milestone-release.md

```diff
@@ -0,0 +1,141 @@
+and the compiler supporting them.
+
+If you’re not familiar with Dotty, it's a platform to try out new language concepts and compiler
+technologies for Scala. The focus is mainly on simplification. We remove extraneous syntax
+(e.g. no XML literals), and try to boil down Scala’s types into a smaller set of more fundamental
+constructors. The theory behind these constructors is researched in
+[DOT](https://infoscience.epfl.ch/record/215280), a calculus for dependent object types.
+You can learn more about Dotty on our [website](http://dotty.epfl.ch).
+
+<!--more-->
+
+This is our fourth scheduled release according to our [6-week release schedule](http://dotty.epfl.ch/docs/usage/version-numbers.html).
+The [previous technology preview](/blog/2017/09/07/third-dotty-milestone-release.html) improved
+stability and reliability.
+
+## What’s new in the 0.4.0-RC1 technology preview?
+
+### Rewritten REPL [#2991](https://github.com/lampepfl/dotty/pull/2991)
+The original Dotty REPL was a proof of concept hacked together
+from
+[an ancient version of the scalac REPL](https://github.com/lampepfl/dotty/pull/1082#issuecomment-183905504).
+It worked by creating Scala source files from the user input using string
+concatenation, this made it easy to adapt it for Dotty since it did not rely on
+the internals of scalac, but it was also fragile and hard to reason about.
+The [new REPL](https://github.com/lampepfl/dotty/pull/2991) instead works by
+manipulating ASTs (Abstract Syntax Trees), this is more robust and will make it
+easier to develop new features: we have already implemented auto-completion
+support (by reusing the APIs we had created for
+the [Dotty IDE](http://dotty.epfl.ch/docs/usage/ide-support.html)) and we have
+plans for displaying API documentation in the REPL.
+
+Note that the user interface of the REPL has not changed: like in the old REPL
+we use code adapted from the [Ammonite REPL](http://ammonite.io/#Ammonite-REPL)
+to provide syntax highlighting, multi-line editing, history, etc.
+
+### Scala 2.12 support [#2827](https://github.com/lampepfl/dotty/pull/2827)
+Since our first release, it has been possible to use Scala 2 libraries in a
+Dotty project as explained in the
+[dotty-example-project](https://github.com/smarter/dotty-example-project#getting-your-project-to-compile-with-dotty).
+Previously, we supported libraries compiled by Scala 2.11, but starting with this
+release we support Scala 2.12 instead. If your Dotty project has Scala 2
+dependencies this change should be transparent for you assuming all your
+dependencies have been published for 2.12.
+
+### Performance work
+Over the last few weeks, we started working on compilation speed with some good results:
+compiling [ScalaPB](https://github.com/dotty-staging/scalapb) is now 20% faster
+than with Dotty 0.3.0-RC2. You can follow along our progress on
+http://dotty-bench.epfl.ch/.
+
+
+## Trying out Dotty
+### Scastie
```

Comment:
Maybe you should link to an already-configured Dotty snippet? Else the default is Scala 2.12.x...

Reply:
Oh, oops, you already do that. I should've looked at the URL for this link. 🙈